cli/types/cypress.d.ts

```diff
@@ -3014,6 +3014,15 @@ declare namespace Cypress {
      * @see https://on.cypress.io/configuration#experimentalModifyObstructiveThirdPartyCode
      */
     experimentalModifyObstructiveThirdPartyCode: boolean
+    /**
+     * Disables setting document.domain to the document's super domain on injection.
+     * This experiment is to be used for sites that do not work with setting document.domain
+     * due to cross-origin issues. Enabling this option no longer allows for default sub domain
+     * navigations, and will require the use of cy.origin().
+     * @see https://developer.mozilla.org/en-US/docs/Web/API/Document/domain
+     * @default false
+     */
+    experimentalUseDefaultDocumentDomain: boolean
```

Comment:
What changes would it take to make this the default behavior for all users (taking it out of experimental)?

Reply:
I think it depends where we land, whether its an "all or nothing" approach like what's implemented right now or something that is configurable per domain. As for now, the only thing we would need to do to take this out of experimental is making the `policyForDomain` always `same-origin` and reverse the conditional which is pretty straight forward. And the change would be breaking.

However I think right now it likely may not work with same origin iframes within the AUT, since we need partial `document.domain` injection for same-origin `localhost` iframes. There might need to be some caveats around this, which would probably be a good idea to document.